### PR TITLE
Feature/agentensor hardening

### PR DESCRIPTION
**Pull request by ShaojieJiang**
Consolidated Agentensor runtime around the vendored engine and wired Orcheo to it.

1. packages/agentensor: added GraphTrainer with prompt snapshots and optional duck-typed optimizer params, exposed core classes via agentensor.__init__, and extended Optimizer to accept direct params/duck-typed modules (train.py, optim.py, init.py).
1. Orcheo AgentensorNode now routes both evaluation and training through GraphTrainer/LLMTensorJudge, uses unified dataset adapter, snapshot-based checkpoints, and syncs trained prompts; evaluator adapter now passes TextTensors to LLMTensorJudge (src/orcheo/nodes/agentensor.py).
1. AgentNode now accepts TextTensor system prompts, advertises trainable params for optimization, and normalizes prompts for create_agent (src/orcheo/nodes/ai.py).
1. Example training judge now subclasses LLMTensorJudge; tests updated to stub LLM calls and cover duck-typed optimizer params (examples/agentensor/train.py, tests/agentensor/*, tests/nodes/test_ai_agent_node_messages.py).
1. Fixed Agentensor training so LLMTensorJudge gets text-like inputs without throwing, and prompt updates are reflected during epoch callbacks. The changes are in agentensor.py, where I added lightweight text wrappers for evaluator contexts and now sync self.prompts on each epoch so progress callbacks see the updated prompt text.

### CODE REVIEW

File: src/orcheo/nodes/agentensor.py [hunk 1]
```diff
@@ -644,13 +764,29 @@ async def _emit_checkpoint(
...
+            prompt_payload: dict[str, Any] = {}
+            for name, prompt in prompt_source.items():
+                if hasattr(prompt, "model_dump"):
+                    prompt_payload[name] = prompt.model_dump(mode="json")  # type: ignore[call-arg]
+                elif isinstance(prompt, TextTensor):
+                    prompt_payload[name] = {
+                        "text": prompt.text,
+                        "type": "TextTensor",
+                        "requires_grad": getattr(prompt, "requires_grad", False),
+                        "metadata": getattr(prompt, "metadata", {}),
```
[the remark this answers is not in the document]
Comment (ShaojieJiang) [reply src/orcheo/nodes/agentensor.py:792]: Updated checkpoint serialization to prefer the synced TrainablePrompt metadata when runtime prompts are TextTensor, so model kwargs aren’t dropped.

1. agentensor.py: _checkpoint_config now checks for a matching trainable prompt when serializing TextTensor entries, preserving model_kwargs (and any future model fields) via model_dump.
1. test_agentensor_node.py: added a regression test to verify model kwargs survive checkpoint serialization after syncing runtime prompts.

**Comment by ShaojieJiang** [comment]
@claude, review this PR

**Comment by ShaojieJiang** [comment]
@claude, review again. I've replaced the bare assertions in the trainer with explicit validation and clearer failure handling, and added strict optimizer parameter validation so misconfigured get_params() surfaces immediately.

1. train.py: added _require_dataset/_require_optimizer checks and a default failure reason when evaluators return no message.
1. optim.py: added _coerce_params to validate params and get_params() outputs with descriptive errors.
1. test_train.py: new coverage for missing dataset/optimizer and default failure reason.
1. test_optim.py: new coverage for invalid params and bad get_params() return types.
